### PR TITLE
Splitting out the jaxrs2 filters

### PR DESCRIPTION
**Pull request by dray92**
Signed-off-by: Debosmit Ray <debo@uber.com>

### CODE REVIEW

File: jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientFilter.java [hunk 1]
```diff
@@ -36,30 +33,22 @@
...
   public ClientFilter(Tracer tracer, TraceContext traceContext) {
     this.tracer = tracer;
     this.traceContext = traceContext;
+
+    this.spanCreationFilter = new ClientSpanCreationFilter(tracer, traceContext);
+    this.spaninjectionFilter = new ClientSpanInjectionFilter(tracer, traceContext);
   }
 
   @Override
   public void filter(ClientRequestContext clientRequestContext) throws IOException {
```
Comment (black-adder) [inline jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientFilter.java]: Just so I follow, for your use case, you'll extend the ClientFilter and then override this filter function to add your baggage between the spanCreationFilter and spanInjectionFilter?

What do we do in the future when other people want to add extra functionality? Will they add their functionality into your filter?

Comment (yurishkuro) [reply jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientFilter.java]: then someone can add a CompositeFilter

Comment (black-adder) [reply jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientFilter.java]: right, so is that the plan?

Comment (yurishkuro) [reply jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientFilter.java]: there's no plan - it's not a problem today, and it's solvable in the future with composite filter, so we don't need to over-engineer today for non-existing need

**Comment by black-adder** [comment]
every commit has to be signed off, you can squash your commits under one with the sign off

File: jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientSpanCreationFilter.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import javax.ws.rs.RuntimeType;
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.ext.Provider;
+import lombok.extern.slf4j.Slf4j;
+
+@Provider
+@ConstrainedTo(RuntimeType.CLIENT)
+@Slf4j
+public class ClientSpanCreationFilter implements ClientRequestFilter {
```
Comment (willsalz) [inline jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientSpanCreationFilter.java]: I don't think that we should be releasing OSS without proper Javadocs.

**Pull request merged**